### DISCUSSION TITLE
@Transaction on mutation not rollback

### DISCUSSION THREAD

**Discussion by lucasrocha2022** [vendurehq/vendure#1648] [Q&A]
Hello, I´m using @Transaction in my mutation like below

```

 @Mutation(() => NewCreditCardInput)
  @Allow(Permission.Authenticated)
  @Transaction('auto')
  newCreditCard(
    @Ctx() ctx: RequestContext,
    @Args('input') newCreditCardInput: NewCreditCardInput
  ) {
    if (ctx.apiType !== 'shop') {
      throw Error('Not allowed');
    }
    return this.walletService.newCreditCard(ctx, newCreditCardInput);
  }
```

          
  In my walletService i do some typeorm repository saves and throw an Error to check the rollback

```ts
const cardWalletEntitySaved = await this.creditCardWalletRepository.save(
        cardWalletEntity
      );
      
      const savedNewCardVerify =
        await this.creditCardWalletVerifyRepository.save(newCardToVerify);
        
        throw new Error('cancelando');    //for test rollback
```
                
I´ve notice that is not rollingback, i´m missing something?
        
Thank you

**Comment by michaelbromley** [comment]
Hi,

For the transaction to be propagated through your various service calls, you'll need to pass the RequestContext object through to all TransactionalConnection `getRepository(ctx, ...)` calls.

**Reply by lucasrocha2022** [reply]
superb, works like a charm!
thanx!
